File: frontend/src/types/movie.ts
export interface MovieInput {
  title: string;
  genre: string;
  rating: number;
  year: number;
}

export interface Movie {
  id: string;
  title: string;
  genre: string;
  rating: number;
  year: number;
  created_at: string;
  updated_at: string;
}

export interface CursorMovieListResponse {
  movies: Movie[];
  next_cursor: string | null;
  has_more: boolean;
  limit: number;
}

export interface GenreStats {
  name: string;
  count: number;
}

export interface YearStats {
  year: number;
  count: number;
}

export interface SummaryStats {
  totalMovies: number;
  averageRating: number;
  topGenres: GenreStats[];
  totalGenres: number;
}

export interface MovieFilters {
  genre?: string;
  min_rating?: number;
  year?: number;
  title?: string;
  cursor?: string;
  limit?: number;
}

File: api/models/movie.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class MovieInput(BaseModel):
    title: str = Field(..., min_length=1)
    genre: str = Field(..., min_length=1)
    rating: float = Field(..., ge=0, le=10)
    year: int = Field(..., ge=1900, le=2100)
    metadata: Optional[Dict[str, Any]] = Field(default_factory=dict)


class Movie(BaseModel):
    id: str
    title: str
    genre: str
    rating: float
    year: int
    created_at: datetime
    updated_at: datetime


class MovieListResponse(BaseModel):
    movies: List[Movie]
    total: int
    page: int
    limit: int


class CursorMovieListResponse(BaseModel):
    movies: List[Movie]
    next_cursor: Optional[str]
    has_more: bool
    limit: int


class GenreStats(BaseModel):
    name: str
    count: int


class YearStats(BaseModel):
    year: int
    count: int


class SummaryStats(BaseModel):
    totalMovies: int
    averageRating: float
    topGenres: List[GenreStats]
    totalGenres: int